### DISCUSSION TITLE
Little Addition to Seeder

### DISCUSSION THREAD

**Discussion by cbka** [spatie/laravel-permission#2147] [Show and tell]
```
<?php

namespace Database\Seeders;

use Illuminate\Database\Seeder;
use Spatie\Permission\Models\Permission;

class PermissionTableSeeder extends Seeder
{
    /**
     * Run the database seeds.
     *
     * @return void
     */
    public function run()
    {
        $permissions = [
            'role-list',
            'role-create',
            'role-edit',
            'role-delete',
            'user-list',
            'user-create',
            'user-edit',
            'user-delete',
            'areale-list',
            'areale-create',
            'areale-edit',
            'areale-delete',
         ];

         foreach ($permissions as $permission) {

       
            if (!Permission::whereName($permission)->first()){
                Permission::create(['name' => $permission]);


            }
         }
    }
}
```

this part 
`if (!Permission::whereName($permission)->first()){`
helps the seeder to work even if other values are already seeded.. simple but not in the tutorials i found...

**Comment by anwarx4u** [comment]
What about -  

`Permission:: updateOrCreate(['name' => $permission]);`

instead of 

`if (!Permission::whereName($permission)->first())
{
                Permission::create(['name' => $permission]);
 }`
            
I am not sure but you may also use upsert also. It will do everything single query..

**Comment by cbka** [comment]
Hi, tbo I have never heard of that method ... But it really does what it should . Your solution is definitely better 😃


Mit freundlichen Grüßen
Christian Becker



mct GmbH
Mühlstr.8, 76229 Karlsruhe

Vertreten durch Christian Becker

Tel. +49 (721) 464712 - 0<tel:+497214647120>

***@***.******@***.***>
www.mct-datensysteme.de<http://www.mct-datensysteme.de/>



------------------------
ACHTUNG: Dies ist eine vertrauliche Nachricht und nur für den Adressaten bestimmt. Es ist nicht erlaubt, diese Nachricht zu kopieren oder Dritten zugänglich zu machen. Sollten Sie diese Nachricht irrtümlich erhalten haben, bitte ich um Ihre Mitteilung per E-Mail oder unter der oben angegebenen Telefonnummer.
------------------------

________________________________
